### PR TITLE
enable eks cluster to deploy to china region

### LINKED ISSUE #278: [FEATURE] Test EKS Cluster in china region
**Is your feature request related to a module? Please describe.**
 `modules/compute/eks`

**Describe the solution you'd like**
Test if IDF EKS module can be deployed in china region

### PR DESCRIPTION
**Pull request by srinivasreddych**
*Issue #, if available:*
Fixes #278 

*Description of changes:*
enable eks cluster to deploy to china region

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: b64c42818e6fdd186ad0aac6ac24378bd3a7fea1
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=4dc77efb-5e16-4dda-bf4d-9e62d26ad6c2%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by dgraeber** [comment]
CHANGLOG ?

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: e0111f057179b51061857f4c8b36ad90dc580b67
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=a6d632b4-8cb1-4f52-adde-fe8441a56123%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]
Thank you!

File: modules/compute/eks/stack.py [hunk 1]
```diff
@@ -681,6 +682,10 @@ def _create_vpc_cni_chart(
         """
         Creates the VPC CNI Helm chart.
         """
+
+        ECR_DOMAIN = fetch_ecr_domain(self.region)
```
Comment (clokep) [inline modules/compute/eks/stack.py:686]: Does CDK provide the [`AWS::URLSuffix` that CloudFormation](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/pseudo-parameter-reference.html#cfn-pseudo-param-partition) allows?

**Review by dgraeber — DISMISSED** [review]

File: modules/compute/eks/stack.py (hunk 1 above)
Comment (srinivasreddych) [reply modules/compute/eks/stack.py:686]: Thanks for spotting it. Let me test and get back. Yes, CDK provides [domain](https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk/Stack.html#aws_cdk.Stack.url_suffix) support

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: IDFModuleTestsInfrastructur-59JoFoVH64CI
* Commit ID: fcde9b82f595147a105ab73f30293c4a3b2c1211
* Result: SUCCEEDED
* [Build Logs](https://mz6i2qapub.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=112cbdbd-d634-42e9-be1b-b74e667a2479%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/compute/eks/stack.py (hunk 1 above)
Comment (srinivasreddych) [reply modules/compute/eks/stack.py]: It worked :)

**Review by dgraeber — APPROVED** [review]

**Review by kukushking — APPROVED** [review]

**Pull request merged**